### ISSUE TITLE
`az account list` - Key Error: 'state'

### ISSUE THREAD

**Issue by gvilarino** [Azure/azure-cli#5846]
Whenever I use `az account list` I get:

```
$  az account list 
'state'
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/site-packages/knack/cli.py", line 194, in invoke
    cmd_result = self.invocation.execute(args)
  File "/usr/local/lib/python2.7/site-packages/azure/cli/core/commands/__init__.py", line 347, in execute
    six.reraise(*sys.exc_info())
  File "/usr/local/lib/python2.7/site-packages/azure/cli/core/commands/__init__.py", line 319, in execute
    result = cmd(params)
  File "/usr/local/lib/python2.7/site-packages/azure/cli/core/commands/__init__.py", line 180, in __call__
    return super(AzCliCommand, self).__call__(*args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/knack/commands.py", line 109, in __call__
    return self.handler(*args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/azure/cli/core/__init__.py", line 418, in default_command_handler
    result = op(**command_args)
  File "/usr/local/lib/python2.7/site-packages/azure/cli/command_modules/profile/custom.py", line 42, in list_subscriptions
    enabled_ones = [s for s in subscriptions if s['state'] == 'Enabled']
KeyError: 'state'

```
---

### Environment summary

Install Method:  `pip`

CLI version (`az --version`):
```
azure-cli (2.0.29)

acr (2.0.22)
acs (2.0.28)
advisor (0.5.0)
appservice (0.1.29)
backup (1.0.7)
batch (3.1.11)
batchai (0.1.6)
billing (0.1.7)
cdn (0.0.13)
cloud (2.0.12)
cognitiveservices (0.1.11)
command-modules-nspkg (2.0.1)
configure (2.0.14)
consumption (0.2.2)
container (0.1.19)
core (2.0.29)
cosmosdb (0.1.19)
dla (0.0.18)
dls (0.0.19)
eventgrid (0.1.11)
eventhubs (0.1.0)
extension (0.0.10)
feedback (2.1.0)
find (0.2.8)
interactive (0.3.17)
iot (0.1.18)
keyvault (2.0.20)
lab (0.0.17)
monitor (0.1.3)
network (2.0.25)
nspkg (3.0.2)
profile (2.0.20)
rdbms (0.1.0)
redis (0.2.11)
reservations (0.1.1)
resource (2.0.25)
role (2.0.20)
servicebus (0.1.0)
servicefabric (0.0.11)
sql (2.0.23)
storage (2.0.27)
vm (2.0.28)

Python location '/usr/local/opt/python@2/bin/python2.7'
Extensions directory '/Users/gvilarino/.azure/cliextensions'

Python (Darwin) 2.7.14 (default, Mar  9 2018, 23:57:12) 
[GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.39.2)]
```

OS version: macOS 10.13.3 (17D102) - High Sierra

Shell Type: Oh-my-ZSH, macOS Terminal

**Comment by yugangw-msft** [comment]
@gvilarino, did you using CLI 1.0 to login in? this is the only case I know where the `state` could be missing.
I should fix this bug soon.

**Comment by gvilarino** [comment]
@yugangw-msft nope. As stated in my issue output, I'm using 2.0.29

**Comment by yugangw-msft** [comment]
@gvilarino, could you please send the `~/.azure/azureProfile.json` to yugangw at microsoft dot com?

**Comment by gvilarino** [comment]
@yugangw-msft sent

**Comment by gvilarino** [comment]
Ok, @yugangw-msft pointed out that my issue was caused because I used Azure CLI v1 in the past, and there were some leftovers in my `~/.azure/azureProfile.json` file that caused the v2 CLI to crash.

Namely, all JSON entries with `management*` were removed ant the CLI started workin properly.

Leaving this here for any future users who come accross this issue.

I'd suggest showing users a warning when a config JSON file doesn't match the expected schema, it would have made this way easier to debug.
